rust/connlib/clients/android/src/lib.rs

```diff
@@ -72,6 +74,19 @@ pub enum CallbackError {
 }
 
 impl CallbackHandler {
+    fn set_new_tun(&self, new_fd: RawFd) {
+        match Tun::with_fd(new_fd) {
+            Ok(tun) => self
+                .session
+                .as_ref()
+                .expect("session must be set once callbacks get called")
+                .set_tun(tun),
+            Err(e) => {
+                tracing::error!("Failed to make new `Tun`: {e}");
```

Comment:
Currently, any error when creating a new `Tun` device is also only bubbled up https://github.com/firezone/firezone/blob/b668f8944bfdc3018c924008ebe1c1b57c6cb912/rust/connlib/tunnel/src/device_channel.rs#L79 and eventually logged https://github.com/firezone/firezone/blob/b668f8944bfdc3018c924008ebe1c1b57c6cb912/rust/connlib/clients/shared/src/eventloop.rs#L82

so this isn't a change in behaviour.

Reply:
If we fail to create the Tun for some reason, will the app silently go on without tunneling any traffic?

Reply:
Yes, that is the behaviour today.